packages/session/src/lib/session.ts

```diff
@@ -0,0 +1,283 @@
+import type { ParseOptions, SerializeOptions } from 'cookie'
+import type { CookieOptions } from '@remix-run/cookie'
+import { Cookie } from '@remix-run/cookie'
+
+import { warnOnce } from './warnings.ts'
+
+/**
+ * An object of name/value pairs to be used in the session.
+ */
+export interface SessionData {
```

Comment:
This feels like an awkward way to achieve type safety for `session.data` because there's no way to define different data in different session cookies. We should either come up with a solution that works for N session cookies on a site or just return `unknown` values out of `session.data` for now.

Reply:
yeah this is carried forward from Remix v1 - happy to ditch it since it's just sort of a TS lie anyway